reconcile/utils/acs/policies.py

```diff
@@ -0,0 +1,125 @@
+from typing import Any, Optional
+
+from pydantic import BaseModel
+
+from reconcile.utils.acs.base import AcsBaseApi
+
+
+class Scope(BaseModel):
+    cluster: str
+    namespace: Optional[str]
+
+
+class PolicyCondition(BaseModel):
+    field_name: str
+    negate: bool
+    values: list[str]
+
+
+class Policy(BaseModel):
+    name: str
+    description: str
+    notifiers: list[str]
+    categories: list[str]
+    severity: str
+    scope: list[Scope]
+    conditions: list[PolicyCondition]
+
+
+class AcsPolicyApi(AcsBaseApi):
+    def list_custom_policies(self) -> list[Any]:
+        # retrieve summary data for each custom policy
+        return [
+            p
+            for p in self.generic_request("/v1/policies", "GET").json()["policies"]
+            if not p["isDefault"]
+        ]
+
+    def get_custom_policies(self) -> list[Policy]:
+        custom_policy_ids = [p["id"] for p in self.list_custom_policies()]
+        # make individual policy requests to obtain further details
+        custom_policies_api_result = [
+            self.generic_request(f"/v1/policies/{pid}", "GET").json()
+            for pid in custom_policy_ids
+        ]
+        formatted_custom_policies: list[Policy] = []
+        for cp in custom_policies_api_result:
+            conditions: list[PolicyCondition] = []
+            for section in cp["policySections"]:
+                for group in section.get("policyGroups", []):
+                    conditions.append(
+                        PolicyCondition(
+                            field_name=group["fieldName"],
+                            values=[v["value"] for v in group["values"]],
+                            negate=group["negate"],
+                        )
+                    )
+
+            formatted_custom_policies.append(
+                Policy(
+                    name=cp["name"],
+                    description=cp["description"],
+                    notifiers=sorted(cp["notifiers"]),
+                    categories=sorted(cp["categories"]),
+                    severity=cp["severity"],
+                    scope=sorted(
+                        [
+                            Scope(
+                                cluster=s["cluster"],
+                                namespace=s["namespace"],
+                            )
+                            for s in cp["scope"]
+                        ],
+                        key=lambda s: s.cluster,
+                    ),
+                    conditions=conditions,
+                )
+            )
+        return formatted_custom_policies
```

Comment:
Can use list comprehension to avoid mutations:

```python
        return [
            Policy(
                name=cp["name"],
                description=cp["description"],
                notifiers=sorted(cp["notifiers"]),
                categories=sorted(cp["categories"]),
                severity=cp["severity"],
                scope=sorted(
                    [
                        Scope(
                            cluster=s["cluster"],
                            namespace=s["namespace"],
                        )
                        for s in cp["scope"]
                    ],
                    key=lambda s: s.cluster,
                ),
                conditions=[
                    PolicyCondition(
                        field_name=group["fieldName"],
                        values=[v["value"] for v in group["values"]],
                        negate=group["negate"],
                    )
                    for section in cp["policySections"]
                    for group in section.get("policyGroups", [])
                ],
            )
            for cp in custom_policies_api_result
        ]
```

Can also extract small functions out like build one `Policy` and build one `PolicyCondition`.

Reply:
[addressed](https://github.com/app-sre/qontract-reconcile/pull/3987/commits/61d5d048dc672974f3e50347096abac100642f0a)